### PR TITLE
Replace memchr dependency with std::iter

### PR DESCRIPTION
**Pull request by Jhowww**
Closes https://github.com/RustPython/RustPython/issues/4481

### CODE REVIEW

File: Lib/test/test_unicode.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ def test_count(self):
 
     def test_find(self):
         string_tests.CommonTest.test_find(self)
-        # test implementation details of the memchr fast path
+        # test implementation details of the std::iter fast path
```
Comment (fanninpm) [inline Lib/test/test_unicode.py:211]: Is this in the CPython source?

Comment (DimitrisJim) [reply Lib/test/test_unicode.py:211]: yeah, its okay if we don't change this to say we use `std::iter`.

**Comment by youknowone** [comment]
I am sorry. I made the wrong issue. memchr is highly optimized so we can't replace it with std::iter.
I thought it was replacable using newly added std::slice::fill, but it wasn't.
I am really sorry to make you work on it.
If you don't mind, can I share you another (real) good first issue for you?

**Comment by Jhowww** [comment]
It's okay, we can move on to a new issue.
Please assign me that new one, thanks.

**Pull request closed without merging**